libs/beacon/tests/dkg/beacon_manager_tests.cpp

```diff
@@ -0,0 +1,378 @@
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "beacon/beacon_manager.hpp"
+#include "core/byte_array/const_byte_array.hpp"
+#include "crypto/ecdsa.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace fetch;
+using namespace fetch::crypto;
+using namespace fetch::crypto::mcl;
+using namespace fetch::dkg;
+using Certificate    = Prover;
+using CertificatePtr = std::shared_ptr<Certificate>;
+using MuddleAddress  = byte_array::ConstByteArray;
+
+struct DkgOutput
+{
+  PrivateKey              private_key;
+  PublicKey               group_public_key;
+  std::vector<PublicKey>  public_key_shares;
+  std::set<MuddleAddress> qual;
+};
+
+TEST(beacon_manager, dkg_and_threshold_signing)
+{
+  bn::initPairing();
```

Comment:
Is this going to do bad stuff if there is multiple tests that call initPairing?

Reply:
No, initPairing can be called multiple times. It is just not thread safe.